rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/TclGradleLauncher.xtend

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.rcp4.tcltestrun
+
+import java.util.concurrent.ConcurrentHashMap
+import javax.inject.Inject
+import org.eclipse.core.resources.IProject
+import org.eclipse.core.runtime.IProgressMonitor
+import org.eclipse.jface.viewers.IStructuredSelection
+import org.gradle.tooling.GradleConnectionException
+import org.gradle.tooling.GradleConnector
+import org.gradle.tooling.ProjectConnection
+import org.gradle.tooling.ResultHandler
+import org.gradle.tooling.events.ProgressEvent
+import org.gradle.tooling.events.ProgressListener
+import org.slf4j.LoggerFactory
+import org.testeditor.dsl.common.ui.utils.ProjectUtils
+import java.util.Map
+
+public class TclGradleLauncher implements TclLauncher {
+
+	static val logger = LoggerFactory.getLogger(TclGradleLauncher)
+	static val GRADLE_TEST_RESULT_FOLDER = "build/test-results"
+
+	@Inject extension ProjectUtils
+
+	private def String elementIdToFileName(String elementId) {
+		'''TEST-«elementId».xml'''
+	}
+
+	override launchTest(IStructuredSelection selection, IProject project, String elementId, IProgressMonitor monitor, Map<String,Object> options) {
+		monitor.beginTask("Test execution: " + elementId, IProgressMonitor.UNKNOWN)
+		val testResultFile = project.createOrGetDeepFolder(GRADLE_TEST_RESULT_FOLDER).getFile(
+			elementId.elementIdToFileName).location.toFile
+		val GradleConnector connector = GradleConnector.newConnector
+		var ProjectConnection connection = null
+		val container = new ConcurrentHashMap<String, Object>()
+		container.putAll(#{RETURN_CODE -> Integer.valueOf(1) as Object, EXPECTED_FILE -> testResultFile})
+		try {
+			val projectFolder = project.location.makeAbsolute.toFile
+			val task = "test"
+			connection = connector.forProjectDirectory(projectFolder).connect();
+			// val BuildEnvironment environment = connection.model(BuildEnvironment).get();
+			connection.newBuild.addProgressListener(new ProgressListener {
```

Comment:
`ProgressListener` has just one method so we can use the shortened Xtend syntax:

```
val build = connection.newBuild => [
   addProgressListener[
      logger.info(...)
   ]
   withArgument(...)
]
build.run(...)
```